jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -3534,18 +3538,19 @@ private static final void validateArgumentsOfCountedLoop(MethodHandle startHandl
 		MethodType bodyType = bodyHandle.type;
 		Class<?> bodyReturnType = bodyType.returnType;
 		Class<?>[] bodyParamTypes = bodyType.arguments;
-		Class<?> bodyParamType1 = bodyParamTypes[0];
 		
 		/* The signature of the body handle must be either (V,I,A...)V or (I,A...)void */
 		if (void.class == bodyReturnType) {
-			if (int.class != bodyParamType1) {
+			if ((0 == bodyParamTypes.length) || (int.class != bodyParamTypes[0])) {
```

Comment:
`(0 == bodyParamTypes.length)` is used on both sides of the if.  Can you pull it out into a local so it's only executed once?